### PR TITLE
Fix BitVec with symbolic offset and fix TranslatorSmtlib.unique thread safety

### PR DESCRIPTION
**Pull request by geohot**
The EVM BYTE opcode with a symbolic offset argument needs the second handler or it asserts (first obviously on the offset == 0)

And I was seeing name collisions on the SMT vars, first I disabled threading to fix, but then added a lock around the unique variable.

### CODE REVIEW

File: manticore/core/smtlib/operators.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def ULE(a, b):
 
 
 def EXTRACT(x, offset, size):
-    if isinstance(x, BitVec):
+    if isinstance(x, BitVec) and not isinstance(offset, BitVec):
```
Comment (feliam) [inline manticore/core/smtlib/operators.py:127]: This is clearly better!
But now I'm thinking that if `offset` happen to be a BitVec then the else will fail to generate a BitVec of `size` bits.

Comment (geohot) [reply manticore/core/smtlib/operators.py:127]: Hmm, fair point. Could call BitVecExtract on the expression with offset 0 and size size?

Comment (feliam) [reply manticore/core/smtlib/operators.py:127]: Hmm I do not really know if the EXTRACT symbol actually supports symbolic size. 
Maybe Operators.EXTRACT needs to force/ request a concrete size. And any user of it must concretize the size before doing the EXTRACT. ... .

Comment (geohot) [reply manticore/core/smtlib/operators.py]: Oh I don't think size can be symbolic, that will fail. But with the new change, offset should be able to.

Comment (feliam) [reply manticore/core/smtlib/operators.py]: Right! Even from the evm.BYTE size if safely concrete.  nvm that.  
https://github.com/trailofbits/manticore/blob/402de84c0686b892c7ef987882f330cad565760e/manticore/platforms/evm.py#L1603-L1606

Comment (geohot) [reply manticore/core/smtlib/operators.py]: Yea, that's the one that was failing with dynamic offset, but size is always 8.

**Comment by feliam** [comment]
Thank you for this. The CI is complaining about not being able to upload coverage info but it should be fixed in master.  I'm merging this. 

Please keep them coming !

**Pull request merged**